doc: Add v18.0.0 release notes (#4955)
dashpay/dash: doc/release-notes.md

Miscellaneous
-------------
A lot of refactoring, code cleanups and other small fixes were done in this release.

v18.0.0 Change log
==================

See detailed [set of changes](https://github.com/dashpay/dash/compare/v0.17.0.3...dashpay:v18.0.0).

Credits
=======

Thanks to everyone who directly contributed to this release:

- AJ ONeal (coolaj86)
- Christian Fifi Culp (christiancfifi)
- dustinface (xdustinface)
- gabriel-bjg
- Holger Schinzel (schinzelh)
- humbleDasher
- Kittywhiskers Van Gogh (kittywhiskers)
- Konstantin Akimov (knst)
- ktechmidas
- linuxsh2
- Munkybooty
- Nathan Marley (nmarley)
- Odysseas Gabrielides (ogabrielides)
- PastaPastaPasta
- pravblockc
- rkarthik2k21
- Stefan (5tefan)
- strophy
- TheLazieR Yip (thelazier)
- thephez
- UdjinM6
- Vijay (vijaydasmp)
- Vlad K (dzutto)

> As well as everyone that submitted issues and reviewed pull requests.

Reviewer: ```suggestion
As well as everyone that submitted issues, reviewed pull requests, helped debug the release candidates, and write DIPs that were implemented in this release. Notable mentions include:

- Samuel Westrich (quantumexplorer)
- Virgile Bartolo
- xkcd
```

Notable changes
===============

Quorum rotation
--------------
InstantSend quorums will now use a new quorum type and a new algorithm for
establishing quorums, see [DIP-0024](https://github.com/dashpay/dips/blob/master/dip-0024.md).

Quorum rotation is activated via a BIP9 style hard fork that will begin

> signalling on ***TODO*** using bit 7. New quorums will start forming in

Author: ```suggestion
signalling on August 15, 2023 using bit 7. New quorums will start forming in
```

Governance
----------
Several improvements have been made to Dash’s DAO governance system.
The governance proposal fee has been reduced from 5 Dash to 1 Dash following
a vote by masternode owners to do so. For improved security and flexibility,
proposal payouts to pay-to-script-hash (P2SH) addresses are now supported.

These changes wil be activated with the DIP0024 hard fork.

Governance proposals can now be viewed in GUI Governance tab (must be enabled
in Preferences first).

Initial Enhanced Hard Fork support
----------------------------------
The masternode hard fork signal special transaction has been added as the first
step in enabling an improved hard fork mechanism. This enhancement enables
future hard forks to be activated quickly and safely without any
“race conditions” if miners and masternodes update at significantly different
speeds. Effectively there will be a masternode signal on chain in addition to
the miner one to ensure smooth transitions. Details of the enhanced hard fork
system are provided in [DIP-0023](https://github.com/dashpay/dips/blob/master/dip-0023.md).

Network improvements
--------------------

> ***TODO***: write some text here

Author: ```suggestion
We implemented and backported implementations of several improvement proposals.
You can read more about implemented changes in the following documents:
```

- [`DIP-0025`](https://github.com/dashpay/dips/blob/master/dip-0025.md): Compressed headers.
- [`BIP 155`](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki): The 'addrv2' and 'sendaddrv2' messages which enable relay of Tor V3 addresses (and other networks).
- [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients.

KeePass support removed
-----------------------
Please make sure to move your coins to a wallet with a regular passphrase.

Wallet changes
--------------

> ***TODO***: write some text here
> - Added support for empty, encrypted-on-creation and watch-only walets.
> - No more `salvagewallet` option in cmd-line and Repair tab in GUI.
> Check the `salvage` command in the `dash-wallet` tool.

Author: ```suggestion
We continued backporting wallet functionality updates. Most notable changes
are:
- Added support for empty, encrypted-on-creation and watch-only wallets.
- Wallets can be created, opened and closed via a GUI menu now.
- No more `salvagewallet` option in cmd-line and Repair tab in GUI. Check the
`salvage` command in the `dash-wallet` tool.
```

Notable changes
===============

Quorum rotation
--------------
InstantSend quorums will now use a new quorum type and a new algorithm for

> establishing quorums, see [DIP-0024](https://github.com/dashpay/dips/blob/master/dip-0024.md).

Reviewer 2: We should not include a broken link. The DIP isn't available quite yet.
```suggestion
establishing quorums. The upcoming DIP-0024 will provide comprehensive details.
```

Quorum rotation is activated via a BIP9 style hard fork that will begin

> signalling on August 15, 2023 using bit 7. New quorums will start forming in

Reviewer: ```suggestion
signalling on August 15, 2022 using bit 7. New quorums will start forming in
```

Author: 🙈

1152-1440 block range after the activation. Any nodes that do not upgrade by
that time will diverge from the rest of the network.

Deterministic InstantSend
-------------------------
Deterministically verifying InstantSend locks at any point in the future has
been added to support Dash Platform. This update introduces versioning to
InstantSend messages and adds quorum information to them. While the previous
design was sufficient for core chain payments, the platform chain will benefit
from this enhanced verification capability. Details about deterministic
InstantSend are provided in [DIP-0022](https://github.com/dashpay/dips/blob/master/dip-0022.md).

Deterministic InstantSend will be activated with the DIP0024 hard fork.

Governance
----------
Several improvements have been made to Dash’s DAO governance system.
The governance proposal fee has been reduced from 5 Dash to 1 Dash following
a vote by masternode owners to do so. For improved security and flexibility,
proposal payouts to pay-to-script-hash (P2SH) addresses are now supported.

> These changes wil be activated with the DIP0024 hard fork.

Reviewer 3: ```suggestion
These changes will be activated with the DIP0024 hard fork.
```

Governance proposals can now be viewed in GUI Governance tab (must be enabled
in Preferences first).

Initial Enhanced Hard Fork support
----------------------------------
The masternode hard fork signal special transaction has been added as the first
step in enabling an improved hard fork mechanism. This enhancement enables
future hard forks to be activated quickly and safely without any
“race conditions” if miners and masternodes update at significantly different
speeds. Effectively there will be a masternode signal on chain in addition to
the miner one to ensure smooth transitions. Details of the enhanced hard fork
system are provided in [DIP-0023](https://github.com/dashpay/dips/blob/master/dip-0023.md).

Network improvements
--------------------

> We implemented and backported implementations of several improvement proposals.

Reviewer 2: ```suggestion
We implemented and backported implementations of several improvement proposals. These enhancements provide bandwidth savings for header synchronization, expand the type of addresses that can be relayed between peers, and introduce a Bloom filter alternative that uses compression to minimize filter size.

```

Reviewer: Can you refer the the PR / DIP for more information in each feature mentioned here?

Author: DIP/BIPs are listed right below this part

Reviewer 2: Yeah, I didn't include them because they are listed in the same order as the links below

You can read more about implemented changes in the following documents:

> - [`DIP-0025`](https://github.com/dashpay/dips/blob/master/dip-0025.md): Compressed headers.

Reviewer 2: This one is in a draft PR (https://github.com/dashpay/dips/pull/113). Unclear when it might be published, but we should remove the link because it will be broken at release.

> - [`DIP-0025`](https://github.com/dashpay/dips/blob/master/dip-0025.md): Compressed headers.

Reviewer 2: This one is in a draft PR (https://github.com/dashpay/dips/pull/113). Unclear when it might be published. Added a link to a gist with it per discussion with @PastaPastaPasta 
```suggestion
- [`DIP-0025`](https://gist.github.com/thephez/6c4c2a7747298e8b3e528c0c4e98a68c): Compressed headers.
```

- [`BIP 155`](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki): The 'addrv2' and 'sendaddrv2' messages which enable relay of Tor V3 addresses (and other networks).
- [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients.

KeePass support removed
-----------------------
Please make sure to move your coins to a wallet with a regular passphrase.

Wallet changes
--------------

> We continued backporting wallet functionality updates. Most notable changes
> are:

Reviewer 2: ```suggestion
We continued backporting wallet functionality updates. Most notably, the GUI now
includes multi-wallet support, which enables working simultaneously with more than
one wallet file. Other notable changes include:
```

- Added support for empty, encrypted-on-creation and watch-only wallets.
- Wallets can be created, opened and closed via a GUI menu now.

> - No more `salvagewallet` option in cmd-line and Repair tab in GUI. Check the
> `salvage` command in the `dash-wallet` tool.

Reviewer 2: ```suggestion
- No more `salvagewallet` option in cmd-line and Repair tab in GUI. Check the
`salvage` command in the `dash-wallet` tool.
- Creating receive addresses in the GUI must now be done using the "Receive" tab
```

Reviewer: Hasn't this been the case since before this version?

Reviewer 2: I don't think so :thinking:

Reviewer 2: Oh, I guess so. Disregard this comment

Author: I can confirm that there is no "Create receiving address" button (or whatever label it had earlier) in v0.17.0.3

Indexes
-------
The transaction index is moved into `indexes/` folder. The migration of the old
data is done on the first run and does not require reindexing. Note that the data
in the old path is removed which means that this change is not backwards
compatible and you'll have to reindex the whole blockchain if you decide to
downgrade to a pre-v18.0.0 version.

Remote Procedure Call (RPC) Changes
-----------------------------------
Most changes here were introduced through Bitcoin backports mostly related to
the deprecation of wallet accounts in DashCore v0.17 and introduction of PSBT
format.

The new RPCs are:
- `combinepsbt`
- `converttopsbt`
- `createpsbt`
- `decodepsbt`
- `deriveaddresses`
- `finalizepsbt`
- `getblockfilter`
- `getdescriptorinfo`
- `getnodeaddresses`
- `getrpcinfo`
- `joinpsbts`
- `listwalletdir`
- `quorum rotationinfo`
- `scantxoutset`
- `submitheader`
- `testmempoolaccept`
- `utxoupdatepsbt`
- `walletcreatefundedpsbt`
- `walletprocesspsbt`

The removed RPCs are:
- `estimatefee`
- `getinfo`
- `getreceivedbyaccount`
- `keepass`
- `listaccounts`
- `listreceivedbyaccount`
- `move`
- `resendwallettransactions`
- `sendfrom`
- `signrawtransaction`

> Changes in existing RPCs introduced through bitcon backports:

Reviewer 2: ```suggestion
Changes in existing RPCs introduced through bitcoin backports:
```

Author: sigh..

Reviewer 2: I didn't catch that one - my spell checker did :smile:

such as `getrawtransaction`, `vsize` has been added and `size` is now
deprecated. `size` will only be returned if `dashd` is started with
`-deprecatedrpc=size`.
- Added `loaded` in mempool related RPCs indicates whether the mempool is fully
loaded or not
- Added `localservicesnames` in `getnetworkinfo` list the services the node
offers to the network, in human-readable form (in addition to an already
existing `localservices` hex string)
- Added `hwm` in `getzmqnotifications`
- `createwallet` can create blank, encrypted or watch-only wallets now.
- Added `private_keys_enabled` in `getwalletinfo`
- Added `solvable`, `desc`, `ischange` and `hdmasterfingerprint` in `getaddressinfo`
- Added `desc` in `listunspent`

Dash-specific changes in existing RPCs:
- Added `quorumIndex` in `quorum getinfo` and `quorum memberof`
- In rpc `masternodelist` with parameters `full`, `info` and `json` the PoS penalty score of the MN will be returned. For `json` parameter, the field `pospenaltyscore` was added.

Please check `help <command>` for more detailed information on specific RPCs.

Command-line options
--------------------

> Mosl changes here were introduced through Bitcoin backports.

Reviewer 3: ```suggestion
Most changes here were introduced through Bitcoin backports.
```

Author: typos.. typos everywhere :D

Wallet changes
--------------
We continued backporting wallet functionality updates. Most notably, the GUI now
includes multi-wallet support, which enables working simultaneously with more than

> one wallet file. Other notable changes include:

Author: Working with multiple wallets in GUI was possible in v17 already, you'd just have to load/unload them via rpc. https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.17.0.2.md#wallet-changes

Reviewer: Hmm, I guess that's a good point... I guess you can drop that commit?
